packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Providers/ModelFactoryProvider.cs

```diff
@@ -0,0 +1,166 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Microsoft.Generator.CSharp.Expressions;
+using Microsoft.Generator.CSharp.Input;
+using Microsoft.Generator.CSharp.Primitives;
+using Microsoft.Generator.CSharp.Snippets;
+using Microsoft.Generator.CSharp.Statements;
+using static Microsoft.Generator.CSharp.Snippets.Snippet;
+
+namespace Microsoft.Generator.CSharp.Providers
+{
+    internal class ModelFactoryProvider : TypeProvider
+    {
+        private readonly IEnumerable<InputModelType> _models;
+
+        public ModelFactoryProvider(IEnumerable<InputModelType> models)
+        {
+            _models = models;
+        }
+
+        protected override string BuildName()
+        {
+            var span = CodeModelPlugin.Instance.Configuration.LibraryName.AsSpan();
+            if (span.IndexOf('.') == -1)
+                return $"{span}ModelFactory";
+
+            Span<char> dest = stackalloc char[span.Length];
+            int j = 0;
+
+            for (int i = 0; i < span.Length; i++)
+            {
+                if (span[i] != '.')
+                {
+                    dest[j] = span[i];
+                    j++;
+                }
+            }
+
+            var prefix =  dest.Slice(0, j).ToString();
+            return $"{prefix}ModelFactory";
```

Comment:
I think we could make this a helper?

Reply:
Since its only used in model factory for now I will leave it here, but we can move it to string helpers maybe if we find more usages for it.